projects/igniteui-angular/src/lib/directives/button/button.directive.ts

```diff
@@ -66,18 +76,26 @@ export class IgxButtonDirective extends DisplayDensityBase {
      * ```
      * @memberof IgxButtonDirective
      */
-    @HostBinding('attr.role') public role = 'button';
+    @HostBinding('attr.role')
+    public role = 'button';
+
     /**
      * Sets the type of the button.
      * ```html
      * <button  igxButton= "icon"></button>
      * ```
      * @memberof IgxButtonDirective
      */
-    @Input('igxButton') set type(value: string) {
-        this._type = value || this._type;
-        this._renderer.addClass(this.nativeElement, `${this._cssClass}--${this._type}`);
+    @Input('igxButton')
+    set type(value: string) {
+        const newValue = value ? value : this._defaultType;
+        if (this._type !== newValue) {
+            this._renderer.removeClass(this.nativeElement, `${this._cssClassPrefix}--${this._type}`);
```

Comment:
We should not remove the class if the old type is empty.

Reply:
This will just try to remove it. If there is no such class it will just do nothing.